File: scratch/inject_karnataka.js
const fs = require('fs');
const newPackages = JSON.parse(fs.readFileSync('scratch/karnataka.json', 'utf8'));

let content = fs.readFileSync('src/pages/PackageDetails.tsx', 'utf8');

const marker = '};\n\nexport function PackageDetails() {';
const index = content.indexOf(marker);

if (index !== -1) {
    let toInsert = '';
    for (const [id, pkg] of Object.entries(newPackages)) {
        toInsert += '  "' + id + '": ' + JSON.stringify(pkg, null, 4).replace(/\n/g, '\n  ') + ',\n';
    }
    
    const newContent = content.substring(0, index) + toInsert + content.substring(index);
    fs.writeFileSync('src/pages/PackageDetails.tsx', newContent, 'utf8');
    console.log('Packages successfully injected!');
} else {
    console.log('Could not find injection point');
}

File: inject_karnataka.py
import re

with open('add_karnataka_code.js', 'r', encoding='utf-8') as f:
    packages_code = f.read()

target_file = r"src/pages/PackageDetails.tsx"
with open(target_file, "r", encoding="utf-8") as f:
    content = f.read()

# First remove any existing 5001-5006 packages if they exist (they don't, but just in case)
new_content = re.sub(r"(\s*'5001':\s*{.*?'5006':\s*{.*?}\s*,?)", "", content, flags=re.DOTALL)

idx = new_content.find('\n};\n\n// Helper normalization function')
if idx != -1:
    final_content = new_content[:idx] + ',\n' + packages_code + '\n};\n\n// Helper normalization function' + new_content[idx + len('\n};\n\n// Helper normalization function'):]
    with open(target_file, "w", encoding="utf-8") as f:
        f.write(final_content)
    print("Successfully injected Karnataka packages.")
else:
    print("Failed to find injection point.")
